### ISSUE TITLE
ALX: a program that multiplies two numbers.

### ISSUE THREAD

**Issue by ahmed149** [ahmed149/learning-programing-in-C#62]
#include <stdio.h>
#include <stdlib.h>

/**
 * main - Prints the multiplication of two numbers, followed by a new line.
 * @argc: The number of arguments supplied to the program.
 * @argv: An array of pointers to the arguments.
 *
 * Return: If the program receives two arguments - 0.
 *         If the program does not receive two arguments - 1.
 */
int main(int argc, char *argv[])
{
	int num1, num2, prod;

	if (argc != 3)
	{
		printf("Error\n");
		return (1);
	}

	num1 = atoi(argv[1]);
	num2 = atoi(argv[2]);
	prod = num1 * num2;

	printf("%d\n", prod);

	return (0);
}

**Comment by ahmed149** [comment]
gcc -Wall -pedantic -Werror -Wextra -std=gnu89 3-mul.c -o mul

**Comment by ahmed149** [comment]
Of course, I'll explain the program step by step without giving you code and provide additional examples as requested.

This program is a C program that multiplies two numbers and prints the result. Here's a breakdown of what's happening:

1. `int main(int argc, char *argv[])`:
   - `argc` stands for "argument count," and it tells you how many arguments (pieces of information) were given to the program when you run it.
   - `argv` stands for "argument vector" and is an array of strings (text) that contains those arguments.

2. `if (argc != 3)`:
   - This condition checks whether the program received exactly three arguments. In C, the program name itself is counted as an argument (so `argc` is at least 1). Therefore, when you want two additional arguments for multiplication, you check if `argc` is not equal to 3.

3. `printf("Error\n");`:
   - If the condition in the previous step is true (meaning the program didn't receive two additional arguments), it prints an error message.

4. `return (1);`:
   - After printing the error message, the program exits with a return value of 1. This typically indicates that something went wrong.

5. `num1 = atoi(argv[1]);` and `num2 = atoi(argv[2]);`:
   - These lines convert the text (strings) from `argv[1]` and `argv[2]` into integer numbers. `atoi` is a function that does this conversion.

6. `prod = num1 * num2;`:
   - Here, it calculates the product of the two numbers.

7. `printf("%d\n", prod);`:
   - Finally, it prints the product of the two numbers.

So, to summarize:
- The program checks if it received exactly three arguments (including the program name).
- If not, it prints an error message and returns 1 to indicate an error.
- If it received three arguments, it multiplies the second and third arguments (converted from strings to integers) and prints the result.

For example:
- If you run the program with `./program 5 7`, it will print `35`, which is the result of 5 multiplied by 7.
- If you run it with fewer or more than three arguments, it will print "Error" and return 1.

I hope this explanation helps! If you have more questions or need further examples, feel free to ask.
